nservicebus/upgrades/nhibernate-saga-index-fix.md

```diff
@@ -0,0 +1,78 @@
+---
+title: NHibernate Persistence - unique constraints on saga mapping tables
+summary: Instructions on how to add missing unique constraints on saga correlation property column for affected versions.
+component: NHibernate
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 6
+---
+
+
+## Summary
+
+This guidance explains how add missing unique constraints on saga correlation property column
+
+- https://github.com/Particular/NServiceBus.NHibernate/issues/280
+
+This issue may cause duplicated rows in saga mapping tables representing single logical saga instance. The duplicated rows are inserted as a result of race condition during saga creation and missing unique constraint on correlation property column.
+
+
+## Compatibility
+
+This issue has been resolved in the following patch versions as defined in the NServiceBus [support policy](support-policy.md):
+
+- NServiceBus.NHibernate 7.2.1
+- NServiceBus.NHibernate 7.1.6
+
+
+If any of the supported affected minor versions (7.2.x or 7.1.x) are used these should be updated to the latest patch release.
+
+
+## Upgrade steps
+
+Steps:
+
+ * For all saga mapping table
+    * Check if it holds any duplicated rows
+    * If any exist manually merge the duplicates
+    * Add unique constraint on correlation property column
+ * Update endpoint to latest patch release
+ * Deploy the new version
+
+### Checking if table hold duplicates on Microsoft SQL Server
+
+Duplicates in saga mapping table can be detected using following query. It requires providing `sagaMappingTableName` and `correlationPropetyColumnName`:
+
+```sql
+declare @sagaMappingTableName nvarchar(max) = ...
+declare @correlationPropretyColumnName nvarchar(max) = ...
+declare @sql nvarchar(max)
+
+select @sql = 'select ' + @correlationPropertyColumnName + ', count(*) as SagaRows from ' + @sagaMappingTableName + ' group by ' + @correlationPropertyColumnName + ' having count(*) > 1'
+exec sp_executeSQL @sql
+
+```
+
+If the query returns any results each row will represent single logical saga instance. First column of the result will show the correlation property value for logical saga instance.
+
+### Checking if table hold duplicates on Oracle
```

Comment:
I'd rather see Oracle and SQL Server subsections under `Checking for duplicate correlation property values` section. This way only SQL snippets would be different.